src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -57,11 +59,82 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                 }
             }
 
+            var controllerType = actionDescriptor.ControllerTypeInfo.AsType();
+            var propertyMetadata = new List<ModelMetadata>();
+            foreach (var property in actionDescriptor.CommonParameters)
+            {
+                var metadata = _modelMetadataProvider.GetMetadataForProperty(
+                    controllerType,
+                    propertyName: property.Name);
```

Comment:
This would also let us get rid of the annoying `var name = mm.BinderName ?? mm.PropertyName` all over the place by moving that to the MBC.